### PR TITLE
Add support for setting up multiple pools

### PR DESCRIPTION
**Pull request by efcasado**

### CODE REVIEW

**Comment by efcasado** [comment]
Pinging @alex-rufo for visibility.

The changes I've introduced are not backwards-compatible. If we were to bump the version of `AMQPPool` in the `Offer` service we would need to do a couple of minor changes to:

- The environment variables used to configure the RMQ endpoint
- Add the pool name when using the `with_channel` function

File: README.md [hunk 1]
```diff
@@ -5,9 +5,17 @@ AMQPPool manages a pool of AMQP channels for you.
 ## Usage
 
 ```elixir
-:ok = AMQPPool.Channel.with_channel(fn channel ->
-  AMQP.Basic.publish(channel, exchange, routing_key, payload)
-end)
+:ok = AMQPPool.Channel.with_channel(:my_pool,
```
Comment (leozilla) [inline README.md:8]: Should we maybe also provide an overload which uses a "default" pool?

File: lib/amqp_pool/pool.ex [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    [
+      username: username(pool),
+      password: password(pool),
+      virtual_host: vhost(pool),
+      port: port(pool),
+      host: host(pool)
+    ]
+  end
+
+  def username(pool) do
```
Comment (leozilla) [inline lib/amqp_pool/pool.ex]: This pattern shows up 4 times, imo its worth to extract it into a function and reuse that.

**Review by leozilla — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (efcasado) [reply README.md:8]: We could definitely do that! Could you create a task for it?

File: lib/amqp_pool/pool.ex (hunk 2 above)
Comment (efcasado) [reply lib/amqp_pool/pool.ex]: Correct! I could go for something like

```elixir
def get_param(param_name, pool, transformation, default) do
  env_var = env_var(pool, param_name)
    param = param(pool, param_name)
    fetch_value(env_var, param, & &1, :undefined)
end
```

File: README.md (hunk 1 above)
Comment (leozilla) [reply README.md:8]: Sure

File: lib/amqp_pool/pool.ex (hunk 2 above)
Comment (efcasado) [reply lib/amqp_pool/pool.ex]: Done!

File: README.md (hunk 1 above)
Comment (leozilla) [reply README.md:8]: I did put it directly into the sportsbook backlog, didnt know where to put it else
https://bluelabs-eu.atlassian.net/browse/CLT-55

I also wanted to assing it to a Component but there is no "amqp_pool" component yet and I also have not the permissions to create one

**Pull request merged**